Tests/FluentAssertions.Specs/Types/PropertyInfoSelectorSpecs.cs

```diff
@@ -355,6 +355,45 @@ public void When_selecting_properties_return_types_it_should_return_the_correct_
                 , typeof(string), typeof(int), typeof(int), typeof(int), typeof(int)
             });
     }
+
+    [Fact]
+    public void When_selecting_properties_with_only_set_accessors_it_should_return_the_applicable_properties()
+    {
+        // Arrange
+        Type type = typeof(TestClassForPropertyInfoSelector);
```

Comment:
🔧 Although you just copied the way we did things before, I'd like us to start using an inline private class that only contains the relevant property(ies), right under the test. This makes the test much more readable and avoids unnecessarily long names like `TestClassForPropertyInfoSelector`. 

Same for the other tests you introduced.

Reply:
If i unterstand it correctly, you want the tests inside the newly created `ThatArePublicOrInternal` class and under each of the tests?

I saw on other tests, that you introduced the "HelperTestClasses" in a seperated file partialling it

Reply:
> If i unterstand it correctly, you want the tests inside the newly created ThatArePublicOrInternal class and under each of the tests?

Yes. If possible, we want our tests to be as self-explanatory as possible. And using reusable test classes spreaded around the code base doesn't help understanding why the number of properties is expected to be "3"

> I saw on other tests, that you introduced the "HelperTestClasses" in a seperated file partialling it

That was a mistake ;-)